compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt

```diff
@@ -542,128 +537,6 @@ public fun KtTypeParameter.requireIdentifier(): String = nameIdentifier?.text
     "unable to determine the name of this type parameter.", element = this
   )
 
-/**
- * Safely resolves a PSI [KtTypeReference], when that type reference may be a generic expressed by a
- * type variable name. This is done by inspecting the class hierarchy to find where the generic type
- * is declared, then resolving *that* reference.
- *
- * For instance, given:
- *
- * ```
- * interface Factory<T> {
- *   fun create(): T
- * }
- *
- * interface ServiceFactory : Factory<Service>
- * ```
- *
- * The KtTypeReference `T` will fail to resolve, since it isn't a type. This function will instead
- * look to the `ServiceFactory` interface, then look at the supertype declaration in order to
- * determine the type.
- *
- * @receiver The class which actually references the type. In the above example, this would be
- *   `ServiceFactory`.
- */
-@ExperimentalAnvilApi
```

Comment:
I moved these into `ClassReference.Psi`. It requires a PSI implementation, hence I used the concrete `Psi` class as new home.

Reply:
Sidenote: @RBusarow did a tremendous job with implementing this quite complicated utilities.

Reply:
aww shucks